rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+including its different execution types, cryptographic foundations, and unique challenges.
+
+### Introducing NSSA: A User-Centric Approach
+
+NSSA gives users control over their privacy settings by introducing shielded (which create a layer of privacy for the outputs and only the necessary details are shared) 
+and deshielded (which reveal private details making them publicly visible) execution types, in addition to the traditional public and private modes. This flexibility allows 
+users to customize their privacy settings to match their unique needs, whether they require high levels of confidentiality or more transparency. In NSSA, the system is divided 
+into two states: public and private. The public state uses an account-based model, while the private state employs a UTXO-based model. Private executions within NSSA utilize 
+UTXO exchanges, ensuring that transaction details remain confidential. The sequencer verifies these exchanges without accessing specific details, enhancing privacy by unlinking 
+sender and receiver identities. Zero-Knowledge Proofs (ZKPs) allow users to prove transaction validity without revealing data, maintaining the integrity and confidentiality of 
+private transactions. UTXOs contain assets such as balances, NFTs, or private storage data, and are stored in plaintext within Sparse Merkle Trees (SMTs) in the private state and 
+as hashes in the public state. This dual-storage approach keeps UTXO details confidential while allowing public verification through hashes, achieving a balance between privacy and transparency.
+
+
+Implementing NSSA introduces unique challenges, particularly in cryptographic implementation and maintaining the integrity of private executions. These challenges are addressed 
+through various solutions such as ZKPs, which ensure transaction validity without compromising privacy, and the dual-storage approach, which maintains confidentiality while enabling 
+public verification. By allowing users to customize their privacy settings, NSSA enhances user experience and promotes wider adoption of private execution platforms. As we move towards 
+a future where user-empowered privacy control is crucial, NSSA provides a flexible and user-centric solution that meets the diverse needs of blockchain users.
+
+### Why NSSA is different than other execution hybrid platforms
+
+In many existing hybrid execution platforms, privacy settings are predefined by developers, often applying a one-size-fits-all approach that does not accommodate the 
+diverse privacy needs of users. These platforms blend public and private states, but the control over privacy remains with the application developers. 
+While this approach is straightforward for developers, who bear the responsibility for any potential privacy leaks, it leaves users with no control over their own privacy settings. 
```

Comment:
Replace:
"While this approach is straightforward for developers, who bear the responsibility for any potential privacy leaks, it leaves users with no control over their own privacy settings."

With:
"While this approach is straightforward for developers (who bear the responsibility for any potential privacy leaks), it leaves users with no control over their own privacy settings."